msgraph-mail/typescript/src/models/attachmentCollectionResponseImpl.ts

```diff
@@ -0,0 +1,103 @@
+import {Attachment} from './attachment';
+import {AttachmentCollectionResponse} from './attachmentCollectionResponse';
+import {createAttachmentFromDiscriminatorValue} from './createAttachmentFromDiscriminatorValue';
+import {AttachmentImpl} from './index';
+import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';
+
+export class AttachmentCollectionResponseImpl implements AttachmentCollectionResponse {
+    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
+    private _additionalData?: Record<string, unknown> | undefined;
+    /** The OdataNextLink property */
+    private _odataNextLink?: string | undefined;
+    /** The value property */
+    private _value?: Attachment[] | undefined;
+    /**
+     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
+     * @returns a Record<string, unknown>
+     */
+    public get additionalData() {
+        return this._additionalData;
+    };
+    /**
+     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
+     * @param value Value to set for the AdditionalData property.
+     */
+    public set additionalData(value: Record<string, unknown> | undefined) {
+        if(value) {
+            this._additionalData = value;
+        }
+    };
+    /**
+     * Instantiates a new AttachmentCollectionResponse and sets the default values.
+     * @param attachmentCollectionResponseParameterValue 
+     */
+    public constructor(attachmentCollectionResponseParameterValue?: AttachmentCollectionResponse | undefined) {
+        this._additionalData = {};
+        this._additionalData = attachmentCollectionResponseParameterValue?.additionalData ? attachmentCollectionResponseParameterValue?.additionalData! : {};
+        this._odataNextLink = attachmentCollectionResponseParameterValue?.odataNextLink;
+        this._value = attachmentCollectionResponseParameterValue?.value;
```

Comment:
here we should map the values we're getting from the JSON notation to the implementation class with something like 

```TypeScript
attachmentCollectionResponseParameterValue?.value?.map(x => new Attachement(x))
```

Otherwise, the mapping to concrete implementations will stop at the first level
(this comment also applies to properties that are single complex values)